Source/Core/Core/IOS/FS/FileSystemProxy.h

```diff
@@ -24,12 +24,50 @@ constexpr FS::Fd INVALID_FD = 0xffffffff;
 class FSDevice : public Device
 {
 public:
+  class ScopedFd
+  {
+  public:
+    ScopedFd(FSDevice* fs, s64 fd, Ticks tick_tracker = {})
+        : m_fs{fs}, m_fd{fd}, m_tick_tracker{tick_tracker}
+    {
+    }
+
+    ~ScopedFd()
+    {
+      if (m_fd >= 0)
+        m_fs->Close(m_fd, m_tick_tracker);
```

Comment:
Should we add an error here like we do for `WriteSystemFile` ?

Reply:
IMO no, it's not really useful. If we are relying on the scoped behavior, we pretty much don't care whether Close() fails, as long as the resource is cleaned up :)